### PR TITLE
[NNBD] Migrating some Material tests

### PR DESCRIPTION
**Pull request by Piinks**
## Description

This migrates several tests from the material library to NNBD.

## Related Issues

dart-lang/sdk#40146
#62886

## Tests

Pass!

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/material/dropdown_form_field_test.dart [hunk 1]
```diff
@@ -171,12 +169,13 @@ void main() {
...
                 }).toList(),
-                onChanged: (String newValue) {
-                  setState(() {
-                    value = newValue;
-                  });
+                onChanged: (String? newValue) {
+                  if (newValue != null)
+                    setState(() {
+                      value = newValue;
+                    });
```
Comment (goderbauer) [inline packages/flutter/test/material/dropdown_form_field_test.dart:176]: Should we just make `value` nullable? It seems that a null value is ok for the DropdownButtonFormField?

File: packages/flutter/test/material/dropdown_form_field_test.dart [hunk 2]
```diff
@@ -677,7 +676,10 @@ void main() {
   testWidgets('DropdownButton onTap callback is called when defined', (WidgetTester tester) async {
     int dropdownButtonTapCounter = 0;
     String value = 'one';
-    void onChanged(String newValue) { value = newValue; }
+    void onChanged(String? newValue) {
+      if (newValue != null)
```
Comment (goderbauer) [inline packages/flutter/test/material/dropdown_form_field_test.dart:680]: Same here.

File: packages/flutter/test/material/dropdown_test.dart [hunk 3]
```diff
@@ -380,8 +379,9 @@ void main() {
 
   testWidgets('Dropdown button with no app', (WidgetTester tester) async {
     String value = 'one';
-    void didChangeValue(String newValue) {
-      value = newValue;
+    void didChangeValue(String? newValue) {
+      if (newValue != null)
```
Comment (goderbauer) [inline packages/flutter/test/material/dropdown_test.dart:383]: and here?

Comment (goderbauer) [reply packages/flutter/test/material/dropdown_test.dart:383]: and elsewhere in this file?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/test/material/dropdown_test.dart (hunk 3 above)
Comment (Piinks) [reply packages/flutter/test/material/dropdown_test.dart:383]: I was wondering the same by the time I got through it. Let me check again. Thanks!

**Pull request merged**